include/IECore/Reader.h

```diff
@@ -37,6 +37,7 @@
 
 #include "IECore/CompoundObject.h"
 #include "IECore/Export.h"
+#include "IECore/FileNameParameter.h"
```

Comment:
I think this is probably the same problem as we had for MemoryAccumulator/CopyContext - there's no explicit destructor for Reader, so MSVC is trying to compile an implicit one at a point it doesn't have the full definition for FileNameParameter. Could you try declaring an explicit destructor at line 120, and then implement that in the Reader.cpp file?

Reply:
Sure thing I'll try this next. This fix is very much a similar one from back when I was originally porting in 2014, but the explicit destructor might just solve it better:
https://github.com/ImageEngine/cortex/issues/337

Reply:
Fixed using an explicit destructor, great!